client/src/pages/CommunityPage/CommunityPageGroup.js

```diff
@@ -4,82 +4,81 @@ import NavBar from "../../components/Navbar/Navbar";
 import people from "assets/images/profile_default.svg";
 import Banner from "../../components/Banner/Banner";
 import InputTextField from "../../components/InputTextField/InputTextField";
+import InputSelect from "../../components/InputSelect/InputSelect";
 import Banner__Community from "assets/images/community_banner.png";
 import ModalContainerJoin from "../../components/Modal/_ModalContainer-join";
 import MediaCard from "../../components/MediaCard/MediaCard";
 import UpcomingEventsCard from "../../components/UpcomingEventsCard/UpcomingEventsCard";
 import Footer from "../../components/Footer/Footer";
+import axios from 'axios';
+
 
 import books from "../../assets/images/media-card-1.png";
 import happiness from "../../assets/images/media-card-2.png";
 import van from "../../assets/images/media-card-3.png";
 import nightSky from "../../assets/images/media-card-4.png";
 import waterfall from "../../assets/images/media-card-5.png";
 import flight from "../../assets/images/media-card-6.png";
-import pic1 from "../../assets/images/profile-picture-1.png";
-import pic2 from "../../assets/images/profile-picture-2.png";
-import pic3 from "../../assets/images/profile-picture-3.png";
-import pic4 from "../../assets/images/profile-picture-4.png";
-import pic5 from "../../assets/images/profile-picture-5.png";
 import ForumContainer from "../../components/ForumPostContainer/ForumPostContainer";
 import Connections from "../../components/Connections/Connections";
-
-let testUsers = [
-  {
-    userId: 1,
-    userFirstName: "Paula",
-    userSurname: "Bannerman",
-    userPic: pic1
-  },
-  {
-    userId: 2,
-    userFirstName: "Jack",
-    userSurname: "Johnson",
-    userPic: pic2
-  },
-  {
-    userId: 3,
-    userFirstName: "Jenny",
-    userSurname: "Jones",
-    userPic: pic3
-  },
-  {
-    userId: 4,
-    userFirstName: "Joan",
-    userSurname: "Rivers",
-    userPic: pic4
-  },
-  {
-    userId: 5,
-    userFirstName: "Freida",
-    userSurname: "Mercury",
-    userPic: pic5
-  },
-  {
-    userId: 6,
-    userFirstName: "Leslie",
-    userSurname: "Knope",
-    userPic: pic5
-  },
-  {
-    userId: 7,
-    userFirstName: "Frankie",
-    userSurname: "Ocean",
-    userPic: pic4
-  }
-];
+import { BASE_URL } from '../../services/constants';
 
 class CommunityPageGroup extends Component {
   constructor(props) {
     super(props);
+    let startGroup
+    !props.id ? startGroup = 1 : startGroup = props.id
     this.state = {
       joinGroup: false,
-      showModal: false
+      showModal: false,
+      groupId: startGroup,
+      groupData: {},
     };
   }
 
+  componentDidMount() {
+    this.getGroup()
+    this.getGroups()
+  }
```

Comment:
These are both going to happen asynchronously. Can you warp these so that you're only updating state once for both of these?

Reply:
@ZakkMan Is it not better that the page loads the current group independently of the Group drop down list, incase one takes longer than another?